source/MaterialXGenShader/DefaultColorManagementSystem.cpp

```diff
@@ -16,20 +16,36 @@ const string DefaultColorManagementSystem::CMS_NAME = "default_cms";
 // DefaultColorManagementSystem methods
 //
 
-string DefaultColorManagementSystem::getImplementationName(const ColorSpaceTransform& transform) const
+ImplementationPtr DefaultColorManagementSystem::getImplementation(const ColorSpaceTransform& transform) const
 {
-    return "IM_" + transform.sourceSpace + "_to_" + transform.targetSpace + "_" + transform.type->getName() + "_" + _target;
+    if (!_document)
+    {
+        throw ExceptionShaderGenError("No library loaded for color management system");
+    }
+
+    TargetDefPtr targetDef = _document->getTargetDef(_target);
+    const StringVec targets = targetDef->getMatchingTargets();
+    for (const string& target : targets)
+    {
+        const string implName = "IM_" + transform.sourceSpace + "_to_" + transform.targetSpace + "_" + transform.type->getName() + "_" + target;
+        ImplementationPtr impl = _document->getImplementation(implName);
+        if (impl)
+        {
+            return impl;
+        }
+    }
+
+    return nullptr;
 }
 
 DefaultColorManagementSystemPtr DefaultColorManagementSystem::create(const string& target)
 {
-    DefaultColorManagementSystemPtr result(new DefaultColorManagementSystem(target));
-    return result;
+    return DefaultColorManagementSystemPtr(new DefaultColorManagementSystem(target));
 }
 
-DefaultColorManagementSystem::DefaultColorManagementSystem(const string& target)
+DefaultColorManagementSystem::DefaultColorManagementSystem(const string& target) :
+    _target(target)
 {
-    _target = createValidName(target);
```

Comment:
I think this was put in on purpose as we create separate files for each source->destination pair. 
One way to preserve this would be to change getImplementation() to compute a "valid name" for each target pair to compare or is it assumed you cannot create an "invalid" target name? 
I think the latter is true since the target mtlx files need to be valid filenames. Not a big thing but maybe worth noting in the spec that target names must be valid names?

Reply:
Yes my thinking here was that if createValidName would change the target name the targetdef would not be found, since the target sent in matches the targetdef. And yes, a target name must always be a valid identifier name.